src/module_loader.rs

```diff
@@ -175,20 +223,49 @@ impl ModuleLoader for RustyLoader {
                 .boxed_local(),
             ),
 
-            _ => ModuleLoadResponse::Sync(Err(anyhow!(
-                "{} imports are not allowed here: {}",
-                module_specifier.scheme(),
-                module_specifier.as_str()
-            ))),
+            _ => {
+                #[cfg(feature = "custom_import")]
+                if inner.import_provider.is_some() {
+                    return ModuleLoadResponse::Async(
+                        async move {
+                            inner
+                                .load(module_specifier, |specifier| {
+                                    let import_provider = inner.import_provider.as_ref().as_ref().unwrap();
+                                    let maybe_referrer = maybe_referrer.clone();
+                                    let requested_module_type = requested_module_type.clone();

```

Comment:
You're cloning these twice 
Not sure you need to clone at all? This block should be able to take ownership no?

Reply:
When I remove those two lines, I get errors. Will get more info on this later.

Reply:
Specifically, if I remove the first clones, I get "lifetime may not live long enough" and if I don't do the second clone, I get a "can't move out of captured variable" which occurs because Option<Url> doesn't have Copy.